Sources/TuistGenerator/Mappers/LibraryResourcesProjectMapper.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// A project mapper that adds support for defining resources in  libraries
+public class LibraryResourcesProjectMapper: ProjectMapping {
+    public init() {}
+
+    public func map(project: Project) throws -> (Project, [SideEffectDescriptor]) {
+        var sideEffects: [SideEffectDescriptor] = []
+        var targets: [Target] = []
+
+        project.targets.forEach { target in
+            let (mappedTargets, targetSideEffects) = mapTarget(target, project: project)
+            targets.append(contentsOf: mappedTargets)
+            sideEffects.append(contentsOf: targetSideEffects)
+        }
+
+        return (project.with(targets: targets), sideEffects)
+    }
+
+    public func mapTarget(_ target: Target, project: Project) -> ([Target], [SideEffectDescriptor]) {
+        guard target.resources.count != 0 else { return ([target], []) }
+        var additionalTargets: [Target] = []
+        var sideEffects: [SideEffectDescriptor] = []
+
+        let bundleName = "\(target.name)Resources"
+        var modifiedTarget = target
+
+        if target.product.isStatic {
+            let resourcesTarget = Target(name: bundleName,
+                                         platform: target.platform,
+                                         product: .bundle,
+                                         productName: nil,
+                                         bundleId: "\(target.bundleId).resources",
+                                         resources: target.resources,
+                                         filesGroup: target.filesGroup)
+            modifiedTarget.resources = []
+            modifiedTarget.dependencies.append(.target(name: bundleName))
+            additionalTargets.append(resourcesTarget)
+        }
+
+        if target.supportsSources {
+            let (filePath, fileDescriptors) = synthesizedFile(bundleName: bundleName, target: target, project: project)
+            modifiedTarget.sources.append((path: filePath, compilerFlags: nil))
+            sideEffects.append(contentsOf: fileDescriptors)
+        }
+
+        return ([modifiedTarget] + additionalTargets, sideEffects)
+    }
+
+    public func synthesizedFile(bundleName: String, target: Target, project: Project) -> (AbsolutePath, [SideEffectDescriptor]) {
+        let filePath = project.path
+            .appending(component: Constants.DerivedDirectory.name)
+            .appending(component: Constants.DerivedDirectory.sources)
+            .appending(component: "Bundle+\(target.name).swift")
+
+        let content: String = """
+        import class Foundation.Bundle
```

Comment:
Discussing with the team they raised a valid point about objc support.

We could either:

- Auto synthesize a project level `.h` and `.m` in case any of the sources are objc

```objc
@interface NSBundle (\(targetName))

+ (NSBundle *)\(targetName)Bundle;

@end 

// ...
```


- Another option to keep the additions minimal is to add another `@objc` annotated method, the one small downside is we would need to make it public to make it visible to objective-c class and via extensions this may bring us back to the conflicts issue, so perhaps we add another dedicated class for objc purposes 

e.g.

```swift
// MARK: - Swift Bundle Accessor 

extension Foundation.Bundle {
// ...
}

// MARK: - Objective-C Bundle Accessor 

@objc
public class \(tagetName)Resources: NSObject {
   @objc public class var bundle: Bundle {
         return .module
   }
}
```

At the usage point:

```objc
#import <MyModule/MyModule-Swift.h>

// ...

- (void)useResources
{
     NSBundle *bundle = [MyModuleResources bundle];
    // ...
}
```